src/utils/provider.ts

```diff
@@ -0,0 +1,117 @@
+import { FallbackProvider, JsonRpcProvider, WebSocketProvider } from 'ethers';
+import { FallbackProviderConfig } from 'ethers/lib.commonjs/providers/provider-fallback';
```

Comment:
I've did not notice we were importing this type from the internal structure of ethers!, given that any update on this might be a breaking change for us, is there any better way we can handle this type ? Mirroring this type with an interface of ours probably ?? or inferring the type.. idk 🤣